lib/opal/rails/template_handler.rb

```diff
@@ -23,7 +23,11 @@ def local_assigns
 
       def assigns
         <<-'RUBY'.strip
-          JSON.parse(@_assigns.to_json).map { |key, val| "@#{key} = #{val.inspect};" }.join
+          if Rails.application.config.opal.auto_assign_instance_variables?(params.slice(:controller, :action))
```

Comment:
This `if` should be outside of the `<<-'RUBY'` heredoc string, even better is to use an early return:

```rb
unless Rails.application.config.opal.auto_assign_instance_variables?(params.slice(:controller, :action))

<<-'RUBY'.strip
…
```